### PR TITLE
Add subscribe to parent comment action menu

### PR DESCRIPTION
**Pull request by srikarparsi**
### Fixed Issues

$ https://github.com/Expensify/Expensify/issues/309277
PROPOSAL: 

### Tests

Subscribing/Unsubscribing to a thread with comments:
1. Comment in a chat and reply to that comment to create a thread
2. Leave the thread and return to the parent report
3. Hover over the comment and press subscribe
4. Ensure the thread report is in your LHN
5. Then hover over the same chat report and press unsubscribe
6. Ensure that the thread report disappears from your LHN

Subscribing to a thread without comments  (Need 2 users, User A and User B):
1. With User B, comment in a chat
2. With User A, subscribe to the comment that was just created
3. With User B, reply to the comment
4. Ensure that the thread shows up in User As LHN and they receive a notification

- [ ] Verify that no errors appear in the JS console

### QA Steps

- [ ] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] If we are not using the full Onyx data that we loaded, I've added the proper selector in order to ensure the component only re-renders when the data it is using changes
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

https://github.com/Expensify/App/assets/48188732/ae190117-681d-4c07-b7cf-6c6be9637cf9

https://github.com/Expensify/App/assets/48188732/9f34ac65-da1d-4dff-9229-ce9aa105a022

</details>

<details>
<summary>Mobile Web - Chrome</summary>

https://github.com/Expensify/App/assets/48188732/a0332669-f809-47f4-bbba-d4b2d9520f31

</details>

<details>
<summary>Mobile Web - Safari</summary>

https://github.com/Expensify/App/assets/48188732/624096f5-2549-48e2-9a53-5816a2a3950e

</details>

<details>
<summary>Desktop</summary>

https://github.com/Expensify/App/assets/48188732/d4e31556-3c53-4234-946c-33ea6b2dd740

</details>

<details>
<summary>iOS</summary>

image

</details>

<details>
<summary>Android</summary>

https://github.com/Expensify/App/assets/48188732/38eb7055-19e2-4758-ae63-32f208046a1a

</details>

### CODE REVIEW

**Comment by srikarparsi** [comment]
Hi @dannymcclain! Do you think you could send me the bell and bell-slash icon whenever you get a chance. I see them in the figma but not sure whether I should use the regular icons or the flattened ones. This is for the subscribe to thread functionality from this [thread](https://expensify.slack.com/archives/C05RECHFBEW/p1694794809980809).

image

**Comment by dannymcclain** [comment]
No prob! 
[Bell Icons.zip](https://github.com/Expensify/App/files/12724518/Bell.Icons.zip)

**Comment by danieldoglas** [comment]
@srikarparsi please create the PR as a DRAFT instead of a full PR until it's ready to review :)

I'll unassign myself from it for now

**Comment by dannymcclain** [comment]
🔔 🔕 
[Bell Icons.zip](https://github.com/Expensify/App/files/12724628/Bell.Icons.zip)

**Comment by srikarparsi** [comment]
Hey @dannymcclain, I believe the bell icons might be just a little too big?
image

The svg code also looks a little different between the bell icons and the other icons.

For reference, this is the svg code for the chatBubble icon:

```
<?xml version="1.0" encoding="utf-8"?>
<!-- Generator: Adobe Illustrator 26.5.0, SVG Export Plug-In . SVG Version: 6.00 Build 0)  -->
<svg version="1.1" id="Layer_1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px"
	 viewBox="0 0 20 20" style="enable-background:new 0 0 20 20;" xml:space="preserve">
<path d="M18.2,14.2c-0.1-0.3,0-0.6,0.2-0.9c1-1.2,1.6-2.7,1.6-4.3c0-4.4-4.5-8-10-8C4.5,1,0,4.6,0,9c0,4.4,4.5,8,10,8
	c0.9,0,1.8-0.1,2.7-0.3c0.2-0.1,0.5,0,0.7,0.1l5.2,2.7c0.4,0.2,0.8-0.1,0.7-0.6L18.2,14.2z"/>
</svg>
```

And this is svg code for the bell icon (I took out the fill: "black" from the zip file you sent)
```
<svg width="20" height="20" viewBox="0 0 20 20" fill="none" xmlns="http://www.w3.org/2000/svg">
<path fill-rule="evenodd" clip-rule="evenodd" d="M4 8.93133C4 9.30295 3.89557 9.6671 3.69861 9.98223L1.67719 13.2165C1.5614 13.4018 1.5 13.6158 1.5 13.8343C1.5 14.4781 2.02189 15 2.66569 15H17.3343C17.9781 15 18.5 14.4781 18.5 13.8343C18.5 13.6158 18.4386 13.4018 18.3228 13.2165L16.3014 9.98223C16.1044 9.6671 16 9.30295 16 8.93133V7C16 3.68629 13.3137 1 10 1C6.68629 1 4 3.68629 4 7V8.93133ZM10 19C8.51353 19 7.27954 17.9189 7.0415 16.5H12.9585C12.7205 17.9189 11.4865 19 10 19Z"/>
</svg>
```

**Comment by dannymcclain** [comment]
@srikarparsi The reason the SVG code looks a little different is because most of our SVGs have been exported from _Illustrator_ instead of Figma, but I've been running into a lot of issues with Illustrator distorting/changing the paths of an icon. So if it's not a problem, let's go with the slightly cleaner SVG code.

As for the size, I think typically all of our icons are designed and exported at `20px` x `20px` and then in contexts like this where they need to be smaller, the size is adjusted on the front end.

**Comment by srikarparsi** [comment]
Talking to @chiragsalian 1 on 1 but wanted to leave a comment on this Github. Subscribing and unsubscribing is working but the childReportNotificationPreference data isn't updating so the right icon isn't being shown in the context action menu. I think we might need to do something [here](https://github.com/Expensify/App/blob/srikar-addSubscribeToParentComment/src/libs/actions/Report.js#L1351) to fetch updated data but I'm not sure if there's a better way to do it. A video of it working without updating the icons is below:

https://github.com/Expensify/App/assets/48188732/007121e4-1b87-4181-83ac-38f1274a339b

**Comment by chiragsalian** [comment]
In my testing, this is failing,
1. UserA adds a comment, then threads the comment.
2. UserB visits the threads, leaves a comment. This makes them auto-subscribe to the thread and it shows up in the LHN.
image
(the second report `test 1` is the threaded report)
4. Now when UserB goes back to parent and hovers over the action the context menu shows the bell as ready to subscribe which is incorrect since its already subscribed.
image

**Comment by chiragsalian** [comment]
Also you've yet to finish author checklist

File: ios/NewExpensify.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -1338,6 +1351,7 @@
 				PRODUCT_BUNDLE_IDENTIFIER = com.expensify.chat.adhoc;
 				PRODUCT_NAME = "New Expensify AdHoc";
 				PROVISIONING_PROFILE_SPECIFIER = chat_expensify_appstore;
+				"PROVISIONING_PROFILE_SPECIFIER[sdk=iphoneos*]" = expensify_chat_adhoc;
```
Comment (chiragsalian) [inline ios/NewExpensify.xcodeproj/project.pbxproj:1354]: these changes don't look related to your PR.

File: ios/tmp.xcconfig [hunk 2]
```diff
@@ -1 +1,9 @@
-
+NEW_EXPENSIFY_URL=https:/$()/new.expensify.com/
+SECURE_EXPENSIFY_URL=https:/$()/secure.expensify.com/
+EXPENSIFY_URL=https:/$()/www.expensify.com/
+EXPENSIFY_PARTNER_NAME=chat-expensify-com
+EXPENSIFY_PARTNER_PASSWORD=e21965746fd75f82bb66
+PUSHER_APP_KEY=268df511a204fbb60884
+USE_WEB_PROXY=false
+ENVIRONMENT=production
+SEND_CRASH_REPORTS=true
```
Comment (chiragsalian) [inline ios/tmp.xcconfig:9]: this also looks unrelated. let me know if I'm mistaken

File: src/libs/actions/Report.js [hunk 3]
```diff
@@ -1333,12 +1335,66 @@ function updateNotificationPreference(reportID, previousValue, newValue, navigat
...
+ * This will subscribe to an existing thread, or create a new one and then subsribe to it if necessary
+ *
+ * @param {String} childReportID The reportID we are trying to open
+ * @param {Object} parentReportAction the parent comment of a thread
+ * @param {String} parentReportID The reportID of the parent
+ * @param {String} prevNotificationPreference The previous notification preference for the child report
+ *
+ */
+function toggleSubscribeToChildReport(childReportID = '0', parentReportAction = {}, parentReportID = '0', prevNotificationPreference) {
+    if (childReportID !== '0') {
```
Comment (chiragsalian) [inline src/libs/actions/Report.js:1366]: this is weird to me. I think we should return early if childReportID is 0. I don't understand why anyone would want to subscribe to a thread prematurely i.e., when a thread does not exist yet.

File: src/pages/home/report/ContextMenu/ContextMenuActions.js [hunk 4]
```diff
@@ -143,6 +143,72 @@ export default [
...
+        shouldShow: (type, reportAction, isArchivedRoom, betas, anchor, isChronosReport, reportID) => {
+            const childReportNotificationPreference = lodashGet(reportAction, 'childReportNotificationPreference', '');
+            const subscribed = childReportNotificationPreference && childReportNotificationPreference !== 'hidden';
+            if (type !== CONTEXT_MENU_TYPES.REPORT_ACTION) {
+                return false;
+            }
+            const isCommentAction = reportAction.actionName === CONST.REPORT.ACTIONS.TYPE.ADDCOMMENT && !ReportUtils.isThreadFirstChat(reportAction, reportID);
+            const isReportPreviewAction = reportAction.actionName === CONST.REPORT.ACTIONS.TYPE.REPORTPREVIEW;
+            const isIOUAction = reportAction.actionName === CONST.REPORT.ACTIONS.TYPE.IOU && !ReportActionsUtils.isSplitBillAction(reportAction);
+            return !subscribed && (isCommentAction || isReportPreviewAction || isIOUAction);
```
Comment (chiragsalian) [inline src/pages/home/report/ContextMenu/ContextMenuActions.js:162]: i think we should not show this icon if the reportAction does not have a child since that does not make sense to me. Like whats the usecase to subscribe to a thread that does not exist?

**Review by chiragsalian — CHANGES_REQUESTED** [review]
Mentioned a test failure for me [above](https://github.com/Expensify/App/pull/27994#issuecomment-1769565219)

File: src/pages/home/report/ContextMenu/ContextMenuActions.js [hunk 5]
```diff
@@ -143,6 +143,72 @@ export default [
...
+
+            ReportActionComposeFocusManager.focus();
+            Report.toggleSubscribeToChildReport(lodashGet(reportAction, 'childReportID', '0'), reportAction, reportID, childReportNotificationPreference);
+        },
+        getDescription: () => {},
+    },
+    {
+        isAnonymousAction: false,
+        textTranslateKey: 'reportActionContextMenu.unsubscribeFromThread',
+        // textTranslateKey: lodashGet(reportAction, 'childReportNotificationPreference', '0'),
```
Comment (chiragsalian) [inline src/pages/home/report/ContextMenu/ContextMenuActions.js:182]: remove commented out text

**Review by chiragsalian — CHANGES_REQUESTED** [review]

**Comment by chiragsalian** [comment]
I explained why my test is failing in your web-E [PR](https://github.com/Expensify/Web-Expensify/pull/39234#discussion_r1364697813)

File: src/libs/actions/Report.js (hunk 3 above)
Comment (srikarparsi) [reply src/libs/actions/Report.js:1373]: Yeah we talked about it briefly and I asked about it [here](https://expensify.slack.com/archives/C05RECHFBEW/p1697146420218079?thread_ts=1694794809.980809&cid=C05RECHFBEW) which led to this change. It's basically like slack where you can `Get notified about new replies`.

**Comment by chiragsalian** [comment]
PR Author Checklist still incomplete so GH actions are failing.

Also shouldn't this PR be on hold for web-e and auth changes? Otherwise if this is merged and goes live won't it be broken?

**Review by chiragsalian — CHANGES_REQUESTED** [review]
Tests work better and the feature looks more solid, but there is still a flow that i found thats broken,
1. UserA sends a message to UserB.
2. UserB click "subscribe to thread" :boom:
The first subscribe to thread click doesn't do anything for me. The response even says the notification preference is hidden.
If i click a second time then it subscribes to the thread.

**Review by chiragsalian — APPROVED** [review]
LGTM and works well.

**Comment by chiragsalian** [comment]
This PR could have been removed from HOLD yesterday.

Also I just noticed you didn't test on all platforms. In future atleast add a C+ reviewer so that they can test it on all platforms.
This PR hasn't had much movement on it for a while so I'm unsure if i should just yolo it without it being tested on all platforms so that this moves ahead sooner 😅 Alright imma yolo it. QA should hopefully catch issues if any.

**Comment by chiragsalian** [comment]
## Reviewer Checklist

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android: Native
    - [ ] Android: mWeb Chrome
    - [ ] iOS: Native
    - [ ] iOS: mWeb Safari
    - [ ] MacOS: Chrome / Safari
    - [ ] MacOS: Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>



</details>

<details>
<summary>Android: mWeb Chrome</summary>



</details>

<details>
<summary>iOS: Native</summary>



</details>

<details>
<summary>iOS: mWeb Safari</summary>



</details>

<details>
<summary>MacOS: Chrome / Safari</summary>



</details>

<details>
<summary>MacOS: Desktop</summary>



</details>

**Comment by chiragsalian** [comment]
Removing HOLD since https://github.com/Expensify/Web-Expensify/pull/39234 is live.

**Comment by chiragsalian** [comment]
Oh look like PR author checklist isn't completed yet. Maybe you are still working on this. Let us know or update accordingly @srikarparsi.

**Comment by srikarparsi** [comment]
Hey @chiragsalian, so there's one problem I found with more testing.

1. Comment on a report
2. Reply to the comment
3. Go into the thread, and then click leave thread
4. Return to the parent report.
5. The unsubscribe icon will show up even if the report is no longer shared.

https://github.com/Expensify/App/assets/48188732/0ff4cd69-a4fe-4d45-9245-9f10efae0127

The reason this occurs is because for reports you're the owner of, the icon is unsubscribe by default if the child report doesn't exist. I can create a fix for this so that the leaveReport command updates the child report notification preference but the alternative is to modify the leaveThread button to set the notification preference to hidden instead of unsharing the report. What do you think?

Also, would this be better as a follow up issue since it's pretty edge case and would require web, auth and app changes if we go the child report notification preference route. And that way I can fix it alongside any other bugs that come up?

**Comment by srikarparsi** [comment]
Running a build for android

**Comment by chiragsalian** [comment]
> for reports you're the owner of, the icon is unsubscribe by default 

yes that's correct. But I don't see why that's the issue. I also don't think this would require a web change. I think it might require an auth change such that when leaveRoom is called, before we unshare the report we check if user is the owner or maybe first commenter and set their notification preference appropriately.

> Also, would this be better as a follow up issue since it's pretty edge case

Sure a follow-up is fine. Just ask mitch also so that we're on the same page with no surprises.

**Review by chiragsalian — APPROVED** [review]
